src/compiling/lexing/reader.rs

```diff
@@ -10,18 +10,18 @@ pub enum ReadMode {
 }
 
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub struct Reader<'a> {
-    pub code: &'a String,
+pub struct Reader {
```

Comment:
And I'd rename `pub` to `pub(crate)` instead

Reply:
what for?

Reply:
To prevent users from importing this structure out of an external module. This could be done in a separate issue